### PR TITLE
[improvement](statistics)Support partition row count return -1 when it is not fully reported.

### PR DESCRIPTION
**Pull request by Jibing-Li**
For a single partition row count, return -1 to planner when its index is not fully reported.

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 1]
```diff
@@ -1581,11 +1582,32 @@ public long getRowCountForIndex(long indexId, boolean strict) {
             if (strict && !index.getRowCountReported()) {
                 return -1;
             }
-            rowCount += (index == null || index.getRowCount() == -1) ? 0 : index.getRowCount();
+            rowCount += index.getRowCount() == -1 ? 0 : index.getRowCount();
         }
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1586]: Is index alway not null?

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1591]: Added check for null

**Review by englefly — APPROVED** [review]

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run cloud_p0

**Comment by Jibing-Li** [comment]
run cloud_p0

**Comment by Jibing-Li** [comment]
run external

**Comment by Jibing-Li** [comment]
run cloud_p0

**Comment by Jibing-Li** [comment]
run cloud_p0

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run cloud_p0

**Review by morrySnow — APPROVED** [review]

**Pull request merged**